modules/ROOT/pages/zero-to-hero.adoc

```diff
@@ -0,0 +1,293 @@
+* https://enterprisecontract.dev/docs/user-guide/main/cli.html[ec cli]
+* https://github.com/containers/skopeo[skopeo] or
+  https://github.com/google/go-containerregistry/tree/main/cmd/crane[crane]
+* https://docs.docker.com/engine/reference/run/[docker] or https://podman.io/[podman] (optional)
+
+== Container Repository
+
+During this guide, image signatures and attestations will be pushed to the container repository
+where the image resides. Using a local registry is ideal for testing purposes. If you have a
+container runtime installed, e.g. https://docs.docker.com/engine/reference/run/[docker] or
+https://podman.io/[podman], you can start one with ease:
+
+[.console-input]
+[source, bash]
+----
+docker run -it -d -p 5000:5000 --restart=always --name registry registry:2
+# or with podman:
+# podman run -it -d -p 5000:5000 --restart=always --name registry registry:2
+----
+
+Alternatively, if you have access to a repository in an existing registry, e.g. quay.io or
+docker.io, then you can just use that.
+
+Set the environment variable `REPOSITORY` accordingly:
+
+[.console-input]
+[source, bash]
+----
+REPOSITORY='localhost:5000/ec-zero-to-hero'
+----
+
+== Container Image
+
+Before signing and attesting an image, we need... well... a container image! Any image will do. You
+can use the ec-cli image if you need one, `quay.io/enterprise-contract/ec-cli:latest`.
+
+Let's copy the image to our test repository. This can be done with a tool such as
+https://github.com/containers/skopeo[skopeo] or
+https://github.com/google/go-containerregistry/tree/main/cmd/crane[crane]:
+
+[.console-input]
+[source, bash]
+----
+crane copy quay.io/enterprise-contract/ec-cli:latest "$REPOSITORY:latest"
+# or with skopeo:
+# skopeo copy docker://quay.io/enterprise-contract/ec-cli:latest \
+#  "docker://$REPOSITORY:latest" --dest-tls-verify=false
+# NOTE: Only set `--dest-tls-verify=false` if you are using a local registry.
+----
+
+== Generate Signining Key
+
+In order to sign and attest the image, we need a signing key.
+
+[.console-input]
+[source, bash]
+----
+cosign generate-key-pair
+----
+
+Enter a new password for the key-pair on the password prompt. For the purpose of this guide, use
+something memorable like `top-secret`. Otherwise, always use a strong secret password.
+
+The command should generate two files in the currently working directory. `cosign.pub` is the public
+key. We will use this later during verification. This can, and should, be distributed freely to
+anyone that may want to verify the image. `cosign.key` is the private key. This is used for signing
+content. Treat this like the password you provided. No one other than the entity responsible for
+singing and attestating images should have access to this.
+
+== Signing the Image
+
+We use cosign to sign our image.
+
+[.console-input]
+[source, bash]
+----
+cosign sign --key cosign.key $REPOSITORY:latest
+----
+
+Enter the key-pair password and `y` to upload a transaction record to Rekor.
```

Comment:
Maybe link to Rekor in case the user hasn't heard of it? https://docs.sigstore.dev/logging/overview/